src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/Cas3UpdateApplicationOffenderNameJobTest.kt

```diff
@@ -0,0 +1,48 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration
+
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.MigrationJobType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationRepository
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.MigrationJobService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.randomStringLowerCase
+
+class Cas3UpdateApplicationOffenderNameJobTest : IntegrationTestBase() {
+  @Autowired
+  lateinit var migrationJobService: MigrationJobService
+
+  @Autowired
+  lateinit var applicationRepository: ApplicationRepository
+
+  @Test
+  fun `all applications have offender name updated`() {
+    val applicationSchema = temporaryAccommodationApplicationJsonSchemaEntityFactory.produceAndPersist {
+      withPermissiveSchema()
+    }
+
+    val probationRegion = probationRegionEntityFactory.produceAndPersist {
+      withApArea(apAreaEntityFactory.produceAndPersist())
+    }
+
+    val user = userEntityFactory.produceAndPersist {
+      withProbationRegion(probationRegion)
+    }
+
+    val temporaryAccommodationApplications = generateSequence {
+      temporaryAccommodationApplicationEntityFactory.produceAndPersist {
+        withApplicationSchema(applicationSchema)
+        withProbationRegion(probationRegion)
+        withCreatedByUser(user)
+        withName(randomStringLowerCase(20))
+      }
+    }.take(50).toList()
+
+    migrationJobService.runMigrationJob(MigrationJobType.cas3ApplicationOffenderName, 1)
+
+    temporaryAccommodationApplications.forEach {
+      val application = applicationRepository.findTemporaryAccommodationApplicationById(it.id)!!
+      Assertions.assertThat(application.name).isEqualTo(it.name)
```

Comment:
would be useful if we check the name is updated in DB is equal to the FirstName, surname comes from offenderService?

Reply:
Tests are changed